lib/source-code/source-code.js

```diff
@@ -447,7 +447,19 @@ class SourceCode extends TokenStore {
         while (currentToken !== finalToken) {
             const nextToken = this.getTokenAfter(currentToken, { includeComments: true });
 
-            if (currentToken.range[1] !== nextToken.range[0]) {
+            if (
+                currentToken.range[1] !== nextToken.range[0] ||
+
+                /*
+                 * For backward compatibility, check speces in JSXText.
+                 * https://github.com/eslint/eslint/issues/12614
+                 */
+                (
+                    nextToken !== finalToken &&
+                    nextToken.type === "JSXText" &&
+                    /\s/u.test(nextToken.value)
+                )
```

Comment:
Thanks for fixing this. I wonder if we should only add this in for the deprecated version? I feel like since it’s part of the `JSXText` node, it shouldn’t actually be treated as whitespace.